packages/build-tools/src/android/expoUpdates.ts

```diff
@@ -53,12 +57,30 @@ export async function androidSetClassicReleaseChannelNativelyAsync(
     throw new Error(`Couldn't find Android manifest at ${manifestPath}`);
   }
 
+  // Store the release channel in a string resource to ensure it is interpreted as a string
+  const stringResourcePath = await AndroidConfig.Strings.getProjectStringsXMLPathAsync(
+    ctx.reactNativeProjectDirectory
+  );
+  const stringResourceObject = await AndroidConfig.Resources.readResourcesXMLAsync({
+    path: stringResourcePath,
+  });
+  const resourceName = 'release_channel';
+  const releaseChannelResourceItem = AndroidConfig.Resources.buildResourceItem({
+    name: resourceName,
+    value: escapedReleaseChannel,
+  });
+  const newStringResourceObject = AndroidConfig.Strings.setStringItem(
```

Comment:
will this work for multi-flavor bare projects that do not have `app/src/main/res` directory? It's a very niche use-case so I'm not saying to support it, but we should at least throw some meaningful error message.

Does it work if release_channel is already specified?

Reply:
verified on my own
- it works even if `res` directory does not exists
- if the release channel is already present it's overridden

Reply:
Good questions and 🙏  thank you for investigating!

This seems like desirable behavior to me.